### PR TITLE
Feature/8347 support tracks in stateless

### PR DESCRIPTION
**Pull request by tjololo**
## Description
Frontend part of https://github.com/Altinn/altinn-studio/issues/8347 
Support for tracks in stateless apps

## Fixes
- https://github.com/Altinn/altinn-studio/issues/8347 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)
- [ ] Changelog is updated with a separate linked PR
  - [ ] [altinn-app-frontend](https://docs.altinn.studio/community/changelog/app-frontend/)- (if applicable)
  - [ ] [nuget-packages](https://docs.altinn.studio/community/changelog/app-nuget/) - (if applicable)

### CODE REVIEW

File: src/altinn-app-frontend/src/utils/appMetadata.test.ts [hunk 1]
```diff
@@ -98,11 +98,11 @@ describe('utils/appmetadata.ts', () => {
...
-      const result = getCurrentDataTypeForApplication(
-        application,
-        instance,
-        layoutSets,
-      );
+      const result = getCurrentDataTypeForApplication({
+        application: application,
+        instance: instance,
+        layoutSets: layoutSets,
+      });
```
Comment (nkylstad) [inline src/altinn-app-frontend/src/utils/appMetadata.test.ts:105]: Can use shorthand as long as the variable name is the same as the field :)
```javascript
const result = getCurrentDataTypeForApplication({
        application,
        instance,
        layoutSets,
      });
```

File: src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts [hunk 2]
```diff
@@ -209,22 +209,32 @@ export function* calculatePageOrderAndMoveToNextPageSaga({ payload: { runValidat
...
+      dataTypeId = getDataTaskDataTypeId(instance.process.currentTask.elementId,
+        state.applicationMetadata.applicationMetadata.dataTypes);
+      if (layoutSets != null) {
+        layoutSetId = getLayoutsetForDataElement(instance, dataTypeId, layoutSets);
+      }
+    }
     const layoutOrder = yield call(
       post,
-      getCalculatePageOrderUrl(),
+      getCalculatePageOrderUrl(appIsStateless),
```
Comment (nkylstad) [inline src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts:232]: Ref. #52, is this endpoint also set up with AllowAnonymous?

File: src/altinn-app-frontend/src/utils/appMetadata.test.ts [hunk 3]
```diff
@@ -112,11 +112,24 @@ describe('utils/appmetadata.ts', () => {
...
       };
-      const result = getCurrentDataTypeForApplication(
-        statelessApplication,
-        null,
-        layoutSets,
-      );
+      const result = getCurrentDataTypeForApplication({
+        application: statelessApplication,
+        instance: null,
+        layoutSets: layoutSets,
```
Comment (nkylstad) [inline src/altinn-app-frontend/src/utils/appMetadata.test.ts:118]:
```suggestion
        layoutSets,
```

File: src/altinn-app-frontend/src/utils/appMetadata.test.ts [hunk 4]
```diff
@@ -112,11 +112,24 @@ describe('utils/appmetadata.ts', () => {
...
+    });
+    
+    it('should return correct data type if instance not set', () => {
+      const statelessApplication = {
+        ...application,
+        onEntry: { show: 'stateless' },
+      };
+      const result = getCurrentDataTypeForApplication({
+        application: statelessApplication,
+        layoutSets: layoutSets,
```
Comment (nkylstad) [inline src/altinn-app-frontend/src/utils/appMetadata.test.ts:131]:
```suggestion
        layoutSets,
```

File: src/altinn-app-frontend/src/features/form/datamodel/fetch/fetchFormDatamodelSagas.ts [hunk 5]
```diff
@@ -24,7 +24,11 @@ function* fetchJsonSchemaSaga(): SagaIterator {
     const instance: IInstance = yield select(InstanceDataSelector);
     const layoutSets: ILayoutSets = yield select((state: IRuntimeState) => state.formLayout.layoutsets);
 
-    const dataTypeId = getCurrentDataTypeForApplication(appMetadata, instance, layoutSets);
+    const dataTypeId = getCurrentDataTypeForApplication({
+      application: appMetadata, 
+      instance: instance, 
+      layoutSets: layoutSets
+    });
```
Comment (nkylstad) [inline src/altinn-app-frontend/src/features/form/datamodel/fetch/fetchFormDatamodelSagas.ts:31]: Can use shorthand declaration when variable name is the same as field name :) 
```javascript
const dataTypeId = getCurrentDataTypeForApplication({
      application: appMetadata, 
      instance, 
      layoutSets
    });
```

File: src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts [hunk 6]
```diff
@@ -209,22 +209,32 @@ export function* calculatePageOrderAndMoveToNextPageSaga({ payload: { runValidat
...
       formData,
       {
         params: {
           currentPage: currentView,
-          layoutSetId,
+          layoutSetId: layoutSetId,
```
Comment (nkylstad) [inline src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts:237]:
```suggestion
          layoutSetId,
```

File: src/altinn-app-frontend/src/utils/appMetadata.ts [hunk 7]
```diff
@@ -61,18 +67,16 @@ export function getLayoutSetIdForApplication(
  * @returns the current data type
  */
 export function getCurrentDataTypeForApplication(
-  application: IApplication,
-  instance?: IInstance,
-  layoutSets?: ILayoutSets,
+  params: IGetDataTypeForApplicationParams
```
Comment (nkylstad) [inline src/altinn-app-frontend/src/utils/appMetadata.ts:70]: Prefer to spread this object directly here, so we avoid using `params` prefix everywhere.
```suggestion
  { application, instance, layoutSets }: IGetDataTypeForApplicationParams
```

**Review by nkylstad — COMMENTED** [review]
Nice! A few comments for you to look at 😄

File: src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts (hunk 2 above)
Comment (tjololo) [reply src/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts:232]: just removed the authorize attribute from stateless endpoint. As the app does not store any data and the endpoint should only return the order of pages we might not need an autorized endpoint for this?

**Review by lorang92 — APPROVED** [review]
Looks good to me! 🥳 

We have cypress tests for existing setup for tracks/sporvalg so I'm comfortable with just merging this and then having @andreas-didriksen verify the new functionality in an app before we close the issue.

Remember to bump package.json version 🥇

File: src/altinn-app-frontend/package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "altinn-app-frontend",
-  "version": "3.36.4",
```
Comment (lorang92) [inline src/altinn-app-frontend/package.json:3]: Should probably patch minor verrsion ❤️

**Pull request merged**